### PR TITLE
Add some tests for shared mutex corner cases

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Will help with validating the behaviors in https://github.com/dotnet/runtime/pull/117635

### CODE REVIEW

File: src/libraries/Common/src/Interop/Unix/Interop.GetPageSize.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+using System.Runtime.InteropServices;
+using System.Threading;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        private static volatile int s_pageSize;
+
+        internal static int PageSize
```
Comment (jkotas) [inline src/libraries/Common/src/Interop/Unix/Interop.GetPageSize.cs]: Is this any different from Environment.SystemPageSize? Can we just use that instead?

Comment (jkotas) [reply src/libraries/Common/src/Interop/Unix/Interop.GetPageSize.cs]: (`SystemNative_GetPageSize` can stay - if you would like to keep the more streamlined implementation that avoids `sysconf` indirection.)

Comment (jkoritzinsky) [reply src/libraries/Common/src/Interop/Unix/Interop.GetPageSize.cs]: We can use that instead for these tests. I didn't realize we had an API for it.

**Review by jkotas — APPROVED** [review]
LGTM once the build breaks are fixed
